### ISSUE TITLE
processDebuggerGetScriptSource in metro-inspector-proxy/src/Device.js is unable to return source

### ISSUE THREAD

**Issue by wood1986** [react/metro#832]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
When I use flipper to debug `index.bundle?platform=android&dev=true&minify=false&app=com.rn0682&modulesOnly=false&runModule=true` without enabling the JavaScript source map, it shows
```
ENOENT: no such file or directory, open '/Users/wood/Desktop/RN0682/http:/localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.rn0682&modulesOnly=false&runModule=true'
```

![image](https://user-images.githubusercontent.com/5212215/174033727-d7d9ed02-2aeb-4723-a83d-2d2083446a16.png)

And I found 

https://github.com/facebook/metro/blob/1a105ea4f91ff385b2fb5f4c5fcfedab3ec630dc/packages/metro-inspector-proxy/src/Device.js#L488-L510

This always throws an exception and makes me unable to access bundled js file.

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**

`git clone https://github.com/wood1986/RN0682`
`yarn install`
`yarn android`
use flipper to debug

**What is the expected behavior?**

It should not read from this fs if metro server is used

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**

**Comment by motiz88** [comment]
I don't think this is the right solution - the *client* should perform this fetch, not the inspector proxy (which is part of the server). This only works based on assumptions on external conditions, e.g. that `http://localhost:8081` on the client device happens to be forwarded to port 8081 on the server.

**Comment by robhogan** [comment]
Closing as fixed by [`6690b393`](https://github.com/react/metro/commit/6690b393cc1812af02f8eb20853504f0a6188a83), released in Metro 0.76.0. `_processDebuggerGetScriptSource` now checks whether the mapped source path is an HTTP URL and fetches it over HTTP, rather than unconditionally reading it from `projectRoot`.

The inspector proxy has since moved out of Metro into `@react-native/dev-middleware`, and that handling moved with it.
